### PR TITLE
Add ability to exclude files/directories 

### PR DESCRIPTION
**Pull request by jasonmccallister**
### Description
One of the major performance hits on Docker Desktop for macOS and Windows is the file sync performance. This add's the ability to exclude directories (and files) from being bind mounted from the host OS to the container in an effort to improve Docker for Desktop performance. Linux users might not see a performance hit, but it should help with file permissions on things that need to be writable in the container (e.g. `storage`, `vendor`). 

We will need to test to verify performance improvements on macOS and Windows but this should help alleviate some negative performance impacts.

This does mean that if the `storage` and `vendor` directories are excluded, the user must manually create those in the container using `nitro ssh <sitename>`.  

For now, the best way to add excludes is to add it manually using `nitro edit` and update to reflect the directories or files tp exclude:

```yaml
  - hostname: europa.nitro
    path: ~/dev/demo-europa-museum
    version: "8.0"
    webroot: web
    xdebug: false
    blackfire: false
    excludes:
      - vendor
      - storage
```

_**Note**: There is no UI to add/edit the excludes but perhaps we could prompt the user during `nitro add`?_

### Related issues
#391 
#409 

### TODOs

- [ ] Update the match logic to check for new files or directories during `nitro apply`
- [ ] Update the `nitro add` logic to include this new capability
- [ ] Consider adding inline docs (links?) to note the "problems" when excluding specific directories (e.g. white page when excluding `vendor` and _Storage is not writable by PHP_ when `storage` is excluded (@mattstein))

### CODE REVIEW

**Comment by jasonmccallister** [comment]
@angrybrad this is ready for review - we are not adding the exlcludes functionality during `nitro add` and should instead document under advanced usage if/when its needed.

**Comment by jasonmccallister** [comment]
Updated the PR description with some benchmarks using the starter blog.

File: command/apply/internal/match/match.go [hunk 1]
```diff
@@ -105,9 +105,24 @@ func Site(home string, site config.Site, container types.ContainerJSON, blackfir
...
+	case true:
+		for _, mount := range container.Mounts {
+			// make sure the only bind mount matches the path
+			if (mount.Type == "bind") && path != mount.Source {
+				return false
+			}
+		}
+	default:
+		// check the number of binds matches the number of container binds (we exclude the user home, certs, and nginx configs since they are volumes)
+		if len(mounts) != len(container.Mounts)-3 {
```
Comment (angrybrad) [inline command/apply/internal/match/match.go:125]: Instead of hard-coding the 3 container binds we're excluding, maybe set those in an array and check the length to make it a bit more clear.

Comment (jasonmccallister) [reply command/apply/internal/match/match.go]: Resolved with https://github.com/craftcms/nitro/pull/412/commits/99fab3a8a817021a08eb97da64b90eff59302337.

File: pkg/config/config.go [hunk 2]
```diff
@@ -290,6 +292,31 @@ func (s *Site) GetAbsPath(home string) (string, error) {
 	return cleanPath(home, s.Path)
 }
 
+// GetBindMounts takes the users home directory and will return a
+// list of bind mounts that checks the excludes on the site.
```
Comment (timkelty) [inline pkg/config/config.go:296]: > takes the users home directory

…you mean current working directory, not home, right?

File: command/apply/internal/match/match.go [hunk 3]
```diff
@@ -14,6 +14,9 @@ import (
 	"github.com/craftcms/nitro/pkg/containerlabels"
 )
 
+// DEFAULT_MOUNTS is the number of mounts for a standard container
+const DEFAULT_MOUNTS = 3
```
Comment (timkelty) [inline command/apply/internal/match/match.go:18]: Expanding on what @angrybrad was saying…
Wonder if it would be more clear to do something like:

```
const DEFAULT_MOUNTS = ["home", "certs", "nginx"]
```

Then later reference `len(DEFAULT_MOUNTS)`?

Same difference, but it is readable w/o the comments.

Comment (jasonmccallister) [reply command/apply/internal/match/match.go:18]: Going to leave as 3, since we might change those in the future.

Comment (timkelty) [reply command/apply/internal/match/match.go:18]: 👍  maybe move the comment about what the 3 mounts are up to the const?

**Comment by jasonmccallister** [comment]
just pushed the flag to add and create.. LGTM!
